internal/grpc/services/gateway/gateway.go

```diff
@@ -49,18 +50,57 @@ type config struct {
 	OCMProviderAuthorizerEndpoint string `mapstructure:"ocmproviderauthorizersvc"`
 	OCMCoreEndpoint               string `mapstructure:"ocmcoresvc"`
 	UserProviderEndpoint          string `mapstructure:"userprovidersvc"`
+	DataGatewayEndpoint           string `mapstructure:"datagateway"`
 	CommitShareToStorageGrant     bool   `mapstructure:"commit_share_to_storage_grant"`
 	CommitShareToStorageRef       bool   `mapstructure:"commit_share_to_storage_ref"`
 	DisableHomeCreationOnLogin    bool   `mapstructure:"disable_home_creation_on_login"`
-	DataGatewayEndpoint           string `mapstructure:"datagateway"`
 	TransferSharedSecret          string `mapstructure:"transfer_shared_secret"`
-	TranserExpires                int64  `mapstructure:"transfer_expires"`
+	TransferExpires               int64  `mapstructure:"transfer_expires"`
 	TokenManager                  string `mapstructure:"token_manager"`
 	// ShareFolder is the location where to create shares in the recipient's storage provider.
 	ShareFolder   string                            `mapstructure:"share_folder"`
 	TokenManagers map[string]map[string]interface{} `mapstructure:"token_managers"`
 }
 
+// sets defaults
+func (c *config) init() {
+	if c.ShareFolder == "" {
+		c.ShareFolder = "MyShares"
+	}
+
+	c.ShareFolder = strings.Trim(c.ShareFolder, "/")
+
+	if c.TokenManager == "" {
+		c.TokenManager = "jwt"
+	}
+
+	// if services address are not specified we used the shared conf
+	// for the gatewaysvc to have dev setups very quickly.
+	c.AuthRegistryEndpoint = sharedconf.GetGatewaySVC(c.AuthRegistryEndpoint)
+	c.StorageRegistryEndpoint = sharedconf.GetGatewaySVC(c.AuthRegistryEndpoint)
+	c.AppRegistryEndpoint = sharedconf.GetGatewaySVC(c.StorageRegistryEndpoint)
+	c.PreferencesEndpoint = sharedconf.GetGatewaySVC(c.AppRegistryEndpoint)
+	c.UserShareProviderEndpoint = sharedconf.GetGatewaySVC(c.UserShareProviderEndpoint)
+	c.PublicShareProviderEndpoint = sharedconf.GetGatewaySVC(c.PublicShareProviderEndpoint)
+	c.OCMShareProviderEndpoint = sharedconf.GetGatewaySVC(c.OCMShareProviderEndpoint)
+	c.OCMInviteManagerEndpoint = sharedconf.GetGatewaySVC(c.OCMInviteManagerEndpoint)
+	c.OCMProviderAuthorizerEndpoint = sharedconf.GetGatewaySVC(c.OCMInviteManagerEndpoint)
+	c.OCMCoreEndpoint = sharedconf.GetGatewaySVC(c.OCMProviderAuthorizerEndpoint)
+	c.UserProviderEndpoint = sharedconf.GetGatewaySVC(c.OCMCoreEndpoint)
```

Comment:
some misaligned assignments here causing regressions

fix is here: https://github.com/cs3org/reva/pull/848